src/components/SignUpDialogueBox.tsx

```diff
@@ -1,24 +1,89 @@
 // React imports
-import React, { useState } from "react";
+import React, { useState, SyntheticEvent } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useMutation, useApolloClient } from '@apollo/react-hooks';
+import gql from 'graphql-tag';
 
-import UploadFile from "./UploadFile";
+import UploadFile from './UploadFile';
 
 // Type definitions
 type DialogueProps = {
   isVolunteer: boolean;
   setIsVolunteer: any;
 };
 
+const REGISTER_VOLUNTEER = gql`
+  mutation registerVolunteer(
+    $username: String!
+    $password: String!
+    $confirmPassword: String!
+  ) {
+    registerVolunteer(
+      username: $username
+      password: $password
+      confirmPassword: $confirmPassword
+    ) {
+      ok
+      errors {
+        path
+        message
+      }
+      token
+      volunteer {
+        id
+      }
+    }
+  }
+`;
+
+const REGISTER_NONPROFIT = gql`
+  mutation registerNonprofit(
+    $username: String!
+    $password: String!
+    $confirmPassword: String!
+    $mission: String!
+    $description: String!
+    $displayName: String!
+    $contact: String!
+  ) {
+    registerNonprofit(
+      username: $username
+      password: $password
+      confirmPassword: $confirmPassword
+      mission: $mission
+      description: $description
+      displayName: $displayName
+      contact: $contact
+    ) {
+      ok
+      errors {
+        path
+        message
+      }
+      token
+      nonprofit {
+        id
+      }
+    }
+  }
+`;
+
 // Structure
 // Header, Text Fields, Radio Selection, Submit
 const SignUpDialogueBox = (props: DialogueProps) => {
-  const [username, setUsername] = useState("");
-  const [password, setPassword] = useState("");
-  const [passwordConfirm, setPasswordConfirm] = useState("");
-  const [mission, setMission] = useState("");
-  const [description, setDescription] = useState("");
-  const [displayName, setDisplayName] = useState("");
-  const [contact, setContact] = useState("");
+  const [username, setUsername] = useState('');
```

Comment:
There's lots of form libraries out there that will do a lot of this for you.

Reply:
Have you used Formik? I'm looking at their [example](https://codesandbox.io/s/8y578o8152) form in TS and I'm confused how they are able to update their state values